scvid/module/base_module.py

```diff
@@ -32,3 +33,21 @@ class BasePyroModule(pyro.nn.PyroModule, BaseModule, metaclass=PyroABCMeta):
     """
     Base module for all scvi-distributed Pyro modules.
     """
+
+
+class PredictMixin(metaclass=ABCMeta):
+    """
+    Abstract mixin class for modules that can perform prediction.
+    """
+
+    @abstractmethod
+    def predict(self, x_ng: torch.Tensor) -> torch.Tensor:
+        """
+        Perform prediction on data tensor.
+
+        Args:
+            x_ng: Data tensor.
+
+        Returns:
+            Prediction tensor.
+        """
```

Comment:
Ended up using `PredictMixin` (@fedorgrab 👏 ), I think it is less code and more flexible rather than creating new abstract subclasses (have to duplicate subclasses for Pyro modules as well).